libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -200,7 +200,7 @@ bool AP_ESC_Telem::get_raw_rpm(uint8_t esc_index, float& rpm) const
 
     const uint32_t now = AP_HAL::micros();
 
-    if (now < rpmdata.last_update_us || now - rpmdata.last_update_us > ESC_RPM_DATA_TIMEOUT_US) {
+    if ((rpmdata.last_update_us > 0) || ((now - rpmdata.last_update_us) > ESC_RPM_DATA_TIMEOUT_US)) {
```

Comment:
This doesn't look right to me

Reply:
`last_update_us` is getting cleared to 0 when it times out, so we have to still check for that to close the hole on the initial loop.